docs/schema.md

```diff
@@ -0,0 +1,346 @@
+# Deployer Schema
+
+The `schema.yaml` file is the way to declare parameter values that end-user needs to provide
+in order to provision the application. For example, the name of the application, the kubernetes
+namespace, service accounts, etc. It follows a more strict subset of JSON schema specifications.
+The UI uses `schema.yaml` to render the form that end-users will interact with to configure and
+deploy the application.
+
+The information provided by the user for properties defined in `schema.yaml` is available at deployment time.
+
+The schema file must exist at the well known location `/data/schema.yaml` in the deployer image.
+
+This is a simple example of a schema.yaml file:
+
+```yaml
+application_api_version: v1beta
+properties:
+  name:
+    type: string
+    x-google-marketplace:
+      type: NAME
+  namespace:
+    type: string
+    x-google-marketplace:
+      type: NAMESPACE
+required:
+- name
+- namespace
+```
+
+Each entry is defined inside properties, and `required` is a list of required parameters.
+We validate that all required fields are provided before starting the deployment.
+
+## Schema and helm chart
+
+There are a few important notes on how the schema interacts with
+the helm chart:
+
+- Each property defined in the schema effectively maps to a helm value as
+  as defined in the chart's `values.yaml` file. This set of properties
+  **should** be a subset of the keys defined in `values.yaml`.
+- User-supplied inputs, as defined by properties in the schema, effectively
+  override the default values in `values.yaml`.
+- The dot annotation can be used to name properties that map to a nested
+  field inside a hierarchical `values.yaml`.
+- There are two required special properties of `x-google-marketplace` types
+  `NAME` and `NAMESPACE` in every schema. These map to the required helm's
+  release name and namespace.
+- The defaults in the schema can be different from and will override the
+  defaults in `values.yaml`.
+
+As an example, let's say we have the following `values.yaml` file:
+
+```yaml
+database:
+  image:
+    name: gcr.io/google/mysql:5.6
+replicas: 3
+```
+
+We can define the following schema:
+
+```yaml
+properties:
```

Comment:
i think we need an `application_api_version: v1beta1` here?

Reply:
This is not meant to be a full schema. It's for demonstrating the relationship between `values.yaml` and `properties`. Add "abbreviated schema" to make it clearer.